### PR TITLE
3rd Party BOM conversion, simpler skip test stuff

### PR DESCRIPTION
**Pull request by akomakom**
Haven't tried to make this daggy yet

* 3rd-party deps moved to:
https://github.com/Terracotta-OSS/terracotta-thirdparty-bom-4.x/blob/main/pom.xml
* Removed skip test insanity
* Relying on forge-parent's system-tests profile that enables only integration tests

Multiple commits for easier reviewing

### CODE REVIEW

File: terracotta-kit/pom.xml [hunk 1]
```diff
@@ -131,7 +149,7 @@ The Initial Developer of the Covered Software is
                   <addClasspath>false</addClasspath>
                 </manifest>
                 <manifestEntries>
-                  <Class-Path>ehcache/lib/ehcache-${ehcache.version}.jar ehcache/lib/slf4j-api-${slf4j.version}.jar toolkit/lib/terracotta-toolkit-runtime-${project.version}.jar</Class-Path>
+                  <Class-Path>ehcache/lib/ehcache-${ehcache.version}.jar ${slf4j.api.ehcache.classpath.path} toolkit/lib/terracotta-toolkit-runtime-${project.version}.jar</Class-Path>
```
Comment (chrisdennis) [inline terracotta-kit/pom.xml:152]: If we're going to go this way would it make sense to get the ehcache jar name via this mechanism too:
```xml
<executions>
  <execution>
    <id>resolve-slf4j-version</id>
    <goals>
      <goal>build-classpath</goal>
    </goals>
    <configuration>
      <includeGroupIds>net.sf.ehcache</includeGroupIds>
      <includeArtifactIds>ehcache</includeArtifactIds>
      <outputProperty>ehcache.classpath.path</outputProperty>
      <prefix>ehcache/lib</prefix>
      <pathSeparator> </pathSeparator>
    </configuration>
  </execution>
</executions>
```

Comment (akomakom) [reply terracotta-kit/pom.xml]: Did it

**Comment by akomakom** [comment]
also found and fixed one difference in dep graph

File: build-parent/pom.xml [hunk 2]
```diff
@@ -332,10 +325,6 @@ The Initial Developer of the Covered Software is
     </pluginManagement>
 
     <plugins>
-      <plugin>
```
Comment (akomakom) [inline build-parent/pom.xml:492]: Inherited

File: build-parent/pom.xml [hunk 3]
```diff
@@ -399,42 +378,6 @@ The Initial Developer of the Covered Software is
...
-                    def target = properties['target']
-                     if (target == null) fail("-Dtarget needs to be set for this profile")
-                    def targetSet = new HashSet(Arrays.asList(target.split(",")))
-                    def projectGroups = new HashSet()
-                    if (project.properties['group'] != null) {
-                      projectGroups.addAll(Arrays.asList(project.properties['group'].split(",")))
-                    }
-                    targetSet.retainAll(projectGroups)
-                    def shouldSkipTest = targetSet.size() == 0
-                    project.properties.should_skip_tests = String.valueOf(shouldSkipTest)
```
Comment (akomakom) [inline build-parent/pom.xml:586]: Converting `-Dtarget=system` vs `-Dtarget=unit`  to simply `-Psystem-tests` vs not.

File: system-tests-parent/pom.xml [hunk 4]
```diff
@@ -378,15 +375,11 @@ The Initial Developer of the Covered Software is
         <tc.core.artifactId>terracotta-ee</tc.core.artifactId>
         <tc.management.war.groupId>com.terracottatech</tc.management.war.groupId>
         <tc.management.war.artifactId>ent-management-tsa-war</tc.management.war.artifactId>
-        <skipTests>false</skipTests>
       </properties>
     </profile>
 
     <profile>
       <id>check-short</id>
-      <properties>
```
Comment (akomakom) [inline system-tests-parent/pom.xml:440]: inherited

File: system-tests-parent/pom.xml [hunk 5]
```diff
@@ -592,63 +581,6 @@ The Initial Developer of the Covered Software is
...
-                  </source>
-                </configuration>
-              </execution>
-            </executions>
-          </plugin>         
-        </plugins>          
-      </build>
-    </profile>    
-
-    <profile>
```
Comment (akomakom) [inline system-tests-parent/pom.xml:684]: Redundant

**Review by GaryWKeim — APPROVED** [review]

**Pull request merged**